이메일 & 닉네임 중복 체크 기능 리팩토링

zlzzlzz2l opened the pull request:
# 이메일 & 닉네임 중복 체크 기능 리팩토링

---

이메일과 닉네임 중복 체크 기능을 리팩토링하였습니다.

---

이메일 중복일 때
![image](https://user-images.githubusercontent.com/48669011/150821464-9c0bfac4-a7a2-40c8-878d-6a1cebdb884c.png)

이메일 중복이 아닐 때
![image](https://user-images.githubusercontent.com/48669011/150821353-1999917c-c3e9-4ed8-b3d1-ca8eaebeaca3.png)

닉네임 중복일 때
![image](https://user-images.githubusercontent.com/48669011/150821288-849c5475-7217-43df-96f6-29b8512df710.png)

닉네임 중복이 아닐 때
![image](https://user-images.githubusercontent.com/48669011/150821251-b1b87c8d-1466-44b0-8e9d-0c316bdbf0a3.png)

zlzzlzz2l committed:
Refactor: #15 - 회원 컨트롤러 이메일 중복 체크 HTTP 상태 코드 수정

zlzzlzz2l committed:
Refactor: #15 - 회원 컨트롤러 닉네임 중복 체크 HTTP 상태 코드 수정

zlzzlzz2l committed:
Refactor: #15 - 회원 컨트롤러 이메일 중복 체크 로직 수정

동작 과정을 체크하기 위해 작성했던 코드를 수정하였습니다.

zlzzlzz2l committed:
Refactor: #15 - 회원 컨트롤러 닉네임 중복 체크 로직 수정

동작 과정을 체크하기 위해 작성했던 코드를 수정하였습니다.

zlzzlzz2l committed:
Refactor: #15 - 회원 서비스 이메일 중복 체크 파라미터명 수정

zlzzlzz2l committed:
Refactor: #15 - 회원 서비스 닉네임 중복 체크 파라미터명 수정

zlzzlzz2l committed:
Refactor: #15 - 회원 서비스 이메일 & 닉네임 중복 체크 메서드 수정

zlzzlzz2l committed:
Refactor: #15 - 회원 리포지토리 이메일 & 닉네임 중복 체크 메서드 선언부 추가

0xe82de commented on src/main/java/com/fourfourfour/damoa/api/member/repository/MemberRepository.java:
```diff
@@ -0,0 +1,9 @@
+import org.springframework.data.jpa.repository.JpaRepository;
+
+public interface MemberRepository extends JpaRepository<Member, Long> {
+    int countByEmail(String memberEmail);
+    int countByNickname(String memberNickname);
```
count 대신 exists를 사용하는 것이 좋을 것 같습니다 :)

[참고 링크](https://velog.io/@pjh612/JPQL%EC%9D%98-count-%EC%84%B1%EB%8A%A5-%EC%9D%B4%EC%8A%88-%EB%B0%8F-exist-%EC%82%AC%EC%9A%A9-%EB%B0%A9%EB%B2%95)

zlzzlzz2l replied:
반영했습니다.

0xe82de commented on src/main/java/com/fourfourfour/damoa/api/member/service/MemberService.java:
```diff
@@ -9,7 +9,7 @@
     void register(MemberDto user);
-    boolean isEmailDuplication(String UserEmail);
-    boolean isNicknameDuplication(String userNickname);
+    boolean isEmailDuplication(String memberEmail);
+    boolean isNicknameDuplication(String memberNickname);
```
변수 이름에 member를 제거해도 괜찮을 것 같습니다. :)

zlzzlzz2l replied:
반영했습니다.

0xe82de commented on src/main/java/com/fourfourfour/damoa/api/member/service/MemberServiceImpl.java:
```diff
@@ -13,6 +14,9 @@
     @Autowired
     private MemberDao userDao;
 
```
MembeDao를 사용하지 않는다면 삭제하는 것도 좋을 것 같아요. :)

zlzzlzz2l replied:
MemberDao를 삭제하면 내부 코드에서 오류가 발생합니다. 

유저 리팩토링을 진행하면서 삭제하도록 하겠습니다.

0xe82de commented on src/main/java/com/fourfourfour/damoa/api/member/service/MemberServiceImpl.java:
```diff
@@ -29,13 +33,13 @@ public void register(MemberDto user) {
-        return userDao.selectCountByEmail(userEmail) != 0;
+    public boolean isEmailDuplication(String memberEmail) {
+        return memberRepository.countByEmail(memberEmail) != 0;
     }
 
     @Override
-    public boolean isNicknameDuplication(String userNickname) {
-        return userDao.selectCountByNickname(userNickname) != 0;
+    public boolean isNicknameDuplication(String memberNickname) {
+        return memberRepository.countByNickname(memberNickname) != 0;
```
spring data jpa 메서드를 exists로 바꾸시면 이 부분도 수정되어야 할 것 같습니다. :)

zlzzlzz2l replied:
반영했습니다.

0xe82de reviewed (changes requested):
리뷰 확인해주세요~ 고생하셨습니다.

zlzzlzz2l committed:
Refactor: #15 - 회원 리포지토리 이메일 & 닉네임 중복 체크 메서드 수정

count -> exists로 변경하였습니다.

zlzzlzz2l committed:
Refactor: #15 - 회원 서비스 이메일 중복 체크 메서드 파라미터명 수정

zlzzlzz2l committed:
Refactor: #15 - 회원 서비스 닉네임 중복 체크 메서드 파라미터명 수정

zlzzlzz2l committed:
Refactor: #15 - 회원 서비스 이메일 중복 체크 메서드 수정

1. 데이터가 존재하는지 확인만 하면 되므로 총 몇개인지 확인하는 count에서 exists로 수정
2. exists가 boolean 타입이므로 리턴값 변경

zlzzlzz2l committed:
Refactor: #15 - 회원 서비스 닉네임 중복 체크 메서드 수정

1. 데이터가 존재하는지 확인만 하면 되므로 총 몇개인지 확인하는 count에서 exists로 수정
2. exists가 boolean 타입이므로 리턴값 변경

yjs1213 reviewed (approved):
고생하셨습니다~
